source/common/runtime/runtime_impl.cc

```diff
@@ -225,8 +225,25 @@ void SnapshotImpl::walkDirectory(const std::string& path, const std::string& pre
       ENVOY_LOG(debug, "reading file: {}", full_path);
       Entry entry;
       entry.string_value_ = Filesystem::fileReadToEnd(full_path);
+
+      // Trim whitespace.
       StringUtil::rtrim(entry.string_value_);
 
+      // Remove any comments. Comments are only allowed at the beginning of the file. A comment
+      // is a line starting with a '#' character. Comments can span multiple lines if each line
+      // starts with a '#' character. Comments are useful for placeholder files with no value.
+      while (!entry.string_value_.empty()) {
```

Comment:
nit: I would probably add a new function `fileReadAllLines` or something which returns an `std::list<std::string>`. That function is very easy to write using C++ std primitives. Then this function becomes a lot clearer to write.

Reply:
Alternatively, could just do a `split` operation, e.g. https://github.com/abseil/abseil-cpp/tree/master/absl/strings.

Reply:
yeah or that. master doesn't currently depend on absl yet, so could also use the split implementation in StringUtil. The small benefit I see to doing a native impl is that it would correctly handle cross platform line endings IIRC.